storage-access.bs

```diff
@@ -162,7 +162,23 @@ When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>ha
 
     ISSUE: "same authority" here is a placeholder for a future concept that allows user agents to perform [=same site=] checks while adhering to additional security aspects such as the presence of a cross-site parent document, see [whatwg/storage#142](https://github.com/whatwg/storage/issues/142#issuecomment-1122147159). In practice, this might involve comparing the [=site for cookies=] or performing a [=same site=] check with the top-level document.
 
-1. [=Queue a global task=] on the [=permissions task source=] given |global| to [=/resolve=] |p| with |global|'s [=environment/has storage access=].
+1. Let |topLevelSite| be the result of [=obtain a site|obtaining a site=] from the [=top-level origin=] of |doc|'s [=relevant settings object=].
+1. Let |embeddedSite| be the result of [=obtain a site|obtaining a site=] from |doc|'s [=Document/origin=].
+1. Run the following steps [=in parallel=]:
+    1. Let |Whether the User Agent Explicitly Allows Unpartitioned Cookie Access| be an algorithm, given a [=tuple=] |tuple| consisting of [=sites=], runs the following steps. This algorithm returns "none", "allow" or "disallow".
+        
+        Note: The "explicit allow" can come from user agent's per-site allow-lists or the user changing global browser settings, etc..
+        
+        1. If user agent does not have explicit settings for unpartitioned cookie access for |tuple|, return "none".
+        1. If user agent's settings explicitly allow unpartitioned cookie access for |tuple|, return "allow".
+        1. If user agent's settings explicitly disallow unpartitioned cookie access for |tuple|, return "disallow".
+    1. Let |explicitSetting| be the result of |Whether the User Agent Explicitly Allows Unpartitioned Cookie Access| with (|topLevelSite|, |embeddedSite|).
+    1. Let |previousPermissionState| be the result of [=getting the current permission state=] given "<a permission><code>storage-access</code></a>" and |global|.
```

Comment:
Hmm I'm still a bit unsure why we need the permission check here, sorry 😳 , shouldn't this be determined by the has storage access bit? Happy to chat about this offline and follow up here.

Reply:
Clarified with Shuran that this is for controlling access where the access may have been revoked by the user but the iframe still has storage access. In that case it looks like we want the user decision to immediately override the cookie access without pending a reload on the iframe or similar.

It might be good to add a non-normative note to that effect?

I should note that things won't work that way if we add storage support in #102 because that will probably exist for the lifetime of the document, unless we add in some mechanism to sever access once permission is revoked (I'm not sure that's a good idea though).